flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphRestartTest.java

```diff
@@ -632,52 +629,6 @@ public void testRestartWithEagerSchedulingAndSlotSharing() throws Exception {
 		}
 	}
 
-	@Test
-	public void testRestartWithSlotSharingAndNotEnoughResources() throws Exception {
```

Comment:
Sorry for figuring out it late. But I think the simplest way to make it work is to use `SimpleSlotProvider`, given that we are testing the `ExecutionGraph` but not `SchedulerImp` here.
`SimpleSlotProvider` will fail immediately with a `NoResourceAvailableException` if there is not enough resources. We can set the slot num to be parallelism - 1 when constructing `SimpleSlotProvider` to keep the same behavior of the tests.

Here's a sample for the change:
`	@Test
	public void testRestartWithSlotSharingAndNotEnoughResources() throws Exception {
		final int numRestarts = 10;
		final int parallelism = 20;

		final SlotSharingGroup sharingGroup = new SlotSharingGroup();

		final JobVertex source = new JobVertex("source");
		source.setInvokableClass(NoOpInvokable.class);
		source.setParallelism(parallelism);
		source.setSlotSharingGroup(sharingGroup);

		final JobVertex sink = new JobVertex("sink");
		sink.setInvokableClass(NoOpInvokable.class);
		sink.setParallelism(parallelism);
		sink.setSlotSharingGroup(sharingGroup);
		sink.connectNewDataSetAsInput(source, DistributionPattern.POINTWISE, ResultPartitionType.PIPELINED_BOUNDED);

		TestRestartStrategy restartStrategy =
			new TestRestartStrategy(numRestarts, false);

		final ExecutionGraph eg = new ExecutionGraphTestUtils.TestingExecutionGraphBuilder(TEST_JOB_ID, source, sink)
			.setSlotProvider(new SimpleSlotProvider(TEST_JOB_ID, parallelism - 1))
			.setRestartStrategy(restartStrategy)
			.setScheduleMode(ScheduleMode.EAGER)
			.build();

		eg.start(mainThreadExecutor);
		eg.scheduleForExecution();

		// the last suppressed restart is also counted
		assertEquals(numRestarts + 1, eg.getNumberOfRestarts());

		assertEquals(JobStatus.FAILED, eg.getState());

		final Throwable t = eg.getFailureCause();
		if (!(t instanceof NoResourceAvailableException)) {
			ExceptionUtils.rethrowException(t, t.getMessage());
		}
	}`

Reply:
Yes and no I would say. If I'm not mistaken then this test should also fail if I set `new SimpleSlotProvider(TEST_JOB_ID, parallelism)` because we don't do proper slot sharing. What one could argue though, is that we don't need to test recovery with slot sharing if we test restart with no slot sharing and slot sharing individually.

Reply:
You are right. I missed that slot sharing part.
Given the only slot provider that supports slot sharing is SchedulerImp, it would be better to take the initial way you suggested, so that the change can be simpler.